src/core/segment_buffers/segment_inventory.ts

```diff
@@ -862,6 +866,7 @@ function guessBufferedEndFromRangeEnd(
       lastSegmentInRange.bufferedEnd = rangeEnd;
     }
     if (!lastSegmentInRange.precizeEnd &&
+        rangeEnd - lastSegmentInRange.end <= MAX_MANIFEST_BUFFERED_START_END_DIFFERENCE &&
```

Comment:
I don't really get why we would wait for the next clock tick to set the precizeEnd to true and the end of rangeEnd in that case.
Why shouldn't we do :
```

log.debug("SI: range end too far from expected end",
          bufferType, rangeEnd, lastSegmentInRange.end);
lastSegmentInRange.bufferedEnd = lastSegmentInRange.end;
if (bufferedEndLooksCoherent(lastSegmentInRange) &&
    rangeEnd - lastSegmentInRange.end <= MAX_MANIFEST_BUFFERED_START_END_DIFFERENCE) {
  lastSegmentInRange.end = rangeEnd;
  lastSegmentInRange.precizeEnd = true;
}
```

Reply:
Maybe I'm confused as this function is a little too complex but I think that this branch will never be taken as:
```ts
if (bufferedEndLooksCoherent(lastSegmentInRange) &&
    rangeEnd - lastSegmentInRange.end <= MAX_MANIFEST_BUFFERED_START_END_DIFFERENCE) {
```
Is already the exact condition of a whole other branch (`log.debug("SI: found true buffered end",`) that we know is not taken when `log.debug("SI: range end too far from expected end",` is encountered (due to a `else if`).

The supplementary condition here was even written to make sure we do not go into that condition on next clock tick